Re-build does not work

[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong

Clean check-out. cmake, build. It works.

Then do a Rebuild (VC++) or `make clean; make` (Ubuntu on WSL). Re-build fails with missing includes.

```
/mnt/d/Workspace/ja2-stracciatella-clean/src/externalized/RustInterface.h:1:10: fatal error: stracciatella.h: No such file or directory
                  #include "stracciatella.h"
                           ^~~~~~~~~~~~~~~~~                                                          
```
or
```
18>      Finished release [optimized + debuginfo] target(s) in 0.34s
18>Building Custom Rule D:/Workspace/ja2-stracciatella/dependencies/lib-stracciatella/CMakeLists.txt
18>D:\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(231,5): warning MSB8065: Custom build for item "D:\Workspace\ja2-stracciatella\_vs\CMakeFiles\eb5fdc63ca1656b9dc5a4454e0892c10\stracciatella.h.rule" succeeded, but specified output "d:\workspace\ja2-stracciatella\_vs\lib-stracciatella\include\stracciatella.h" has not been created. This may cause incremental build to work incorrectly.
18>Done building project "stracciatella-build.vcxproj".
19>------ Rebuild All started: Project: ja2-launcher, Configuration: RelWithDebInfo x64 ------
```
`lib-stracciatella/include/` is cleared but stracciatella.h is not re-generated. It works again if `dependencies/lib-stracciatella/target/` is deleted, forcing a rust rebuild.

[//]: # (For bugs, please fill in this table [otherwise delete it]. The lines don't have to stay aligned.)

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  master  |  N/A |    no    |    yes     |

[//]: # (Please also include:)
[//]: # (- how to reproduce the bug and, if possible, a save game)
[//]: # (- the operating system)
[//]: # (- in case you know, for enhancements, if this feature was already added by some other source mod like 1.13)

[//]: # (When pasting backtraces or any other logs with hashed enumerations, please format it as a code block.)
[//]: # (You can do that by selecting it and clicking the "<>" icon or manually putting three backticks [```] at each end.)

--- a/rust/stracciatella_c_api/build.rs
+++ b/rust/stracciatella_c_api/build.rs
@@ -16,6 +16,7 @@ fn stracciatella_h() {
     cbindgen::generate(&crate_dir)
         .expect("Unable to generate bindings")
         .write_to_file(&header_path);
+    println!("cargo:rerun-if-changed={:?}", &header_path);
 }
 
 fn main() {

